app/helpers/ems_container_helper/textual_summary.rb

```diff
@@ -71,7 +71,12 @@ def textual_port
   end
 
   def textual_zone
-    {:label => _("Managed by Zone"), :icon => "pficon pficon-zone", :value => @record.zone.name}
+    zone = if @record.zone.visible?
```

Comment:
```
if @record.zone.visible?
elsif @record.zone == Zone.maintenance_zone
end
``` 
is better, the same for other helpers below

Reply:
Hm, what other cases can happen? Because this seems to me as fixing possible backend issues on the frontend :disappointed:

Reply:
Your code doesn't cause error, but invisible zone won't have `zone_before_pause` relation set. 

Current condition could confuse other developers that if `zone.visible` is false, then `zone_before_pause` is always there.